python/src/vmaf/tools/reader.py

```diff
@@ -35,6 +35,7 @@ def __init__(self, filepath, width, height, yuv_type):
 
         self._asserts()
 
+        # TODO python3: this doesn't work well with python3, need to refactor this class
         self.file = open(self.filepath, 'rb')
```

Comment:
This wasn't great in python2 either, and becomes quite hard to follow with the changes around how files are handled in python3

If you're familiar with `YuvReader`, please refactor this to not use things like this

Reply:
I'll try to refactor YuvReader. Which tests does it break in python3?

Reply:
`test/extra/noref_feature_extractor_extratest.py`, they're marked with:
```
@unittest.skipIf(sys.version_info > (3,), reason="TODO python3: unclear how YuvReader is supposed to work")
```